src/network/NetworkController.test.ts

```diff
@@ -160,6 +162,8 @@ describe('NetworkController', () => {
     expect(controller.state.provider.type).toBe('mainnet');
     expect(controller.state.provider.ticker).toBe('ETH');
     expect(controller.state.isCustomNetwork).toBe(false);
+    expect(controller.state.provider.rpcTarget).toBeUndefined();
```

Comment:
Hmm, for this to be a valid test we would want to ensure these _were not_ undefined to start with. I think this would pass before and after your changes.

Could you set the RPC target first, then set the provider time and make this check (maybe in a separate test)? That should test that they're cleared properly, because setting the RPC target would populate these provider config fields.

Reply:
I agree completely, added a new test now that will cover that use case! Thank you!